eng/common/TestResources/setup-az-modules.yml

```diff
@@ -20,8 +20,26 @@ steps:
     displayName: (MacOS) Grant access to ~/.Azure
     condition: contains(variables['OSVmImage'], 'mac')
 
-  # New-TestResources command requires Az module
-  - pwsh: Install-Module -Name Az -Scope CurrentUser -AllowClobber -Force -Verbose
+  # This alternative invocation is being setup so that PowerShell will use
+  # TLS 1.2 when downloading from the PowerShell Gallery. We are seeing instances
+  # of failed downloads similar to reported in this article and it seemed this
+  # this helped resolve the issue. I've adding it as a conditioned step because
+  # I'm not confident that this is the solution ... so I'd like to test it on a few
+  # pipelines prior to making it the default.
+  # Article: https://techcommunity.microsoft.com/t5/windows-powershell/failed-downloading-az-and-other-modules-for-powershell/m-p/1292985
+
+
+  # New-TestResources command requires Az module with TLS 1.2
+  - pwsh: |
+      [Net.ServicePointManager]::SecurityProtocol = [Net.SecurityProtocolType]::Tls12
+      Install-Module -Name Az -Scope CurrentUser -AllowClobber -Force -Verbose
+    condition: and(succeeded(), eq(variables.UseTls12, true))
+    displayName: Install Azure PowerShell module (TLS 1.2)
+
+  # New-TestResources command requires Az module with system default TLS version
+  - pwsh: |
+      Install-Module -Name Az -Scope CurrentUser -AllowClobber -Force -Verbose
```

Comment:
The `-Force` avoids the prompt you'll get. If you want to use what's on the box, either trust they are always there and don't install the module, or - if you want to make sure - use `get-module -list <mod-name>` to see if what you want is installed and only do `install-module -force <mod-name>` otherwise.

Reply:
Yes I think we should check to see if the version we need is installed and if so do nothing and if not do the install.

Reply:
Assuming you even need to do that. If the images are supposed to have this image, does it say what version? It's not like we're using anything new.

Reply:
Yes it says https://github.com/actions/virtual-environments/blob/main/images/win/Windows2019-Readme.md#azure-powershell-modules. It looks like 4.7 on linux and windows and 4.6.1 on mac so we could just try to use it or potentially add an assertion to verify that we have a min version we need.

Reply:
Sounds like a better outcome. Also side steps the whole source reliability issue neatly.

Reply:
@mitchdenny do you plan to pivot this PR to do that work?

Reply:
No I'll log and issue and close this one out.

Reply:
We already have the issue at https://github.com/Azure/azure-sdk-tools/issues/1051.